### PR TITLE
Remove callback functionality from symbol search

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes [AB#1750570](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1750570)

Symbol search is a bit of an odd-duck remote service.  It is kicked off by VS, but keeps a long running callback channel open just to send log messages to.  This has added complexity to the system and definitely makes it confusing in terms of the lifetimes of these objects and how cancellation works.  We've gotten hits from users in the past where the oop side tries to talk to the VS side and ends up failing, causing a yellow bar.

Now, the only reason we had the backchannel was for logging purposes.  However, I'm the owner of this area, and i have never once looked at the actual logs produced to figure out anything.  Indeed, that the logs were on the VS side was an active problem when dealing with OOP crashes since there was no log present in the memory dump to look at to see what had happened prior to the crash.

This change just simplifies things by moving the logging to be on the OOP side only.  Now there's no need for any backchannel back to VS, and the system lifetime/connection is much easier to reason about.

### CODE REVIEW

File: src/Workspaces/Core/Portable/SymbolSearch/SymbolSearchLogService.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.SymbolSearch
+{
+    internal sealed class SymbolSearchLogService : ISymbolSearchLogService
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/SymbolSearch/SymbolSearchLogService.cs:11]: this is just a move of the existing service.  but the cleanup and simplification makes it look new.

File: src/VisualStudio/Core/Def/SymbolSearch/VisualStudioSymbolSearchService.LogService.cs [hunk 2]
```diff
@@ -1,62 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/SymbolSearch/VisualStudioSymbolSearchService.LogService.cs:1]: old service which got simplified and now just lives in the actual process we do the index updating in.

File: src/Workspaces/Remote/ServiceHub/Services/SymbolSearchUpdate/RemoteSymbolSearchUpdateService.cs [hunk 3]
```diff
@@ -65,14 +42,14 @@ public ValueTask<ImmutableArray<PackageWithTypeResult>> FindPackagesWithTypeAsyn
 
         public ValueTask<ImmutableArray<PackageWithAssemblyResult>> FindPackagesWithAssemblyAsync(string source, string assemblyName, CancellationToken cancellationToken)
         {
-            return RunServiceAsync(cancallationToken =>
+            return RunServiceAsync(cancellationToken =>
                 _updateEngine.FindPackagesWithAssemblyAsync(source, assemblyName, cancellationToken),
```
Comment (sharwell) [inline src/Workspaces/Remote/ServiceHub/Services/SymbolSearchUpdate/RemoteSymbolSearchUpdateService.cs:46]: 🤦‍♂️

File: src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs [hunk 4]
```diff
@@ -78,12 +78,11 @@ public async Task TestSessionWithNoSolution()
 
             var service = workspace.Services.GetRequiredService<IRemoteHostClientProvider>();
 
-            var mock = new MockLogService();
             var client = await service.TryGetRemoteHostClientAsync(CancellationToken.None);
 
-            using var connection = client.CreateConnection<IRemoteSymbolSearchUpdateService>(callbackTarget: mock);
+            using var connection = client.CreateConnection<IRemoteSymbolSearchUpdateService>(callbackTarget: null);
```
Comment (sharwell) [inline src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs:83]: ❔ Would we normally omit the `callbackTarget` argument at this point?

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs [hunk 5]
```diff
@@ -75,21 +74,20 @@ public ValueTask UpdateContinuouslyAsync(string source, string localSettingsDire
...
+            return new Updater(this, source, localSettingsDirectory).UpdateInBackgroundAsync(cancellationToken);
         }
 
         private sealed class Updater
         {
             private readonly SymbolSearchUpdateEngine _service;
             private readonly string _source;
             private readonly DirectoryInfo _cacheDirectoryInfo;
-            private readonly ISymbolSearchLogService _logService;
+            private readonly ISymbolSearchLogService _logService = SymbolSearchLogService.Instance;
```
Comment (sharwell) [inline src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:85]: 💭 Might as well get rid of the interface at this point?

**Review by sharwell — APPROVED** [review]

File: src/Workspaces/Remote/ServiceHub/Services/SymbolSearchUpdate/RemoteSymbolSearchUpdateService.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/ServiceHub/Services/SymbolSearchUpdate/RemoteSymbolSearchUpdateService.cs:46]: Found by teh spell checker.  I sent them feedback and kudos how this helped us!

File: src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs:83]: non-optional parameter :)

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:85]: Yeah, agreed.  Will do.

**Pull request merged**